### DISCUSSION TITLE
[Kavita] OPDS folder image URL

### DISCUSSION THREAD

**Discussion by MnsieurJF** [Kareadita/Kavita#2823] [Ideas]
### Idea Description

Hello, 

I use the iOS app Panels to browse and read through my Kavita library. Both work great together.

On the Library view in Panels, all series show the default folder icon and not the cover image I set on Kavita web interface.

I contacted Panels devs and they said it should work since they are using the image URL transited via OPDS.

Possible to pass down the image of all folders (serie, editor, library) via OPDS feed ?

Thank you

![IMG_3545](https://github.com/Kareadita/Kavita/assets/165181180/e46dec4a-11c6-48a3-b750-8297df1f4461)

### Idea Category

API

### Duration of Using Kavita

A few months

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
I'll talk with Panels dev. This is working correctly in all the other clients I've tested:
![Screenshot_20240328-054556](https://github.com/Kareadita/Kavita/assets/735851/18c9facd-2b77-4b72-ae24-4b0d6a79e20b)

**Comment by majora2007** [comment]
I followed up with Panels via their support:
https://discord.com/channels/1103436296153747522/1222860332767121429/1222860332767121429

**Reply by MnsieurJF** [reply]
Thanks for the quick reply, I will track it there

**Comment by MnsieurJF** [comment]
So to keep track of it, we got it working! It was Panels-side. The URL was not read correctly, it missed the « /kavita » part.

Now after seeing the OPDS feed, I stumbled on something that got me wondering: how is the summary valorized? Where does the data come from?
When I open a the information panel on a comic issue in Kavita reader there is a description but it doesn’t seem to be passed on via the OPDS feed.
I’ll attach a screenshot of Kavita and a txt file of the OPDS feed.
[14.txt](https://github.com/Kareadita/Kavita/files/14849077/14.txt)
![IMG_3580](https://github.com/Kareadita/Kavita/assets/165181180/f68c6e1e-6b0a-4b0b-a827-8822320dea50)

**Comment by majora2007** [comment]
The description is in the OPDS feed for the series level. I don't think you can actually put it per issue if I recall.

I have seen very mixed results on apps using it. OPDS is one of the worst implemented standards I've seen. Everyone has different implementations.

![Screenshot_20240328-054556](https://github.com/Kareadita/Kavita/assets/735851/7435971d-4e86-4c8b-b4ca-dd256c89298a)

**Reply by MnsieurJF** [reply]
Is there any way to put it in the `<content>` under the `<entry>` balise?
Currently it’s this :
`<content type="text">application/x-cbz</content>`

You can look at this [OPDS feed](https://github.com/Kareadita/Kavita/files/14853244/14.txt) for example.

Recently Panels implemented a feature that displays these information. It reads:
![IMG_3581](https://github.com/Kareadita/Kavita/assets/165181180/29cb8f10-7292-4157-ac9a-4c4da983f086)

**Reply by majora2007** [reply]
While panels might label as description, I believe other apps use this to help with downloading and support. I can look, but I highly doubt it can be coded with the description. The OPDS spec just isn't flexible enough for rich metadata imo.

**Reply by MnsieurJF** [reply]
Yes it is not easy. Sorry I don’t know anything about it but I made some research and found the [OPDS current implementation specs](https://specs.opds.io/opds-1.2#the-atomentry-element).

There are a few examples where the description is put in the content under entry and there is this :

> OPDS Catalog Entries should use atom:summary and/or atom:content to describe the Publication should not use dc:description or dc:abstract.

I might be wrong, you can tell me you have the most experience with this standard.
I think that’s why Panels went this way.

Do you think it’s worth it to continue to dig this and maybe get it touch with Panels devs?

**Reply by majora2007** [reply]
I was able to do it. https://github.com/Kareadita/Kavita/pull/2832

**Comment by majora2007** [comment]
Closing this as it's not a feature request but a bug and has been solved.
